client/Panels/Topics/Topics.steps.tsx

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+import {
+  And,
+  Given,
+  When,
+  Then,
+  Fusion,
+  Before,
+  After,
+} from 'jest-cucumber-fusion';
+import {
+  fireEvent,
+  render,
+  RenderResult,
+  waitFor,
+} from '@testing-library/react';
+import { Topics } from '.';
+import React, { ReactElement } from 'react';
+import { withApolloProviderReturning } from 'utils/test';
+import {
+  mockGetTopicsRequests,
+  mockGetTopicsResponses,
+} from './useTopicsModel.assets';
+
+let renderResult: RenderResult;
+let component: ReactElement;
+
+Before(() => {
+  jest.useFakeTimers();
+});
+
+After(() => {
+  jest.useRealTimers();
+});
+
+Given('a Topics panel component', () => {
+  component = <Topics />;
+});
+
+When('it is rendered', () => {
+  renderResult = render(
+    withApolloProviderReturning(
+      [
+        mockGetTopicsRequests.successRequest,
+        mockGetTopicsRequests.successFilteredRequest,
+      ],
+      component
+    )
+  );
+  jest.runAllTimers();
+});
+
+And(/^I filter for topic '(.+)'$/, async (filter) => {
+  fireEvent.change(renderResult.getByPlaceholderText('filter'), {
+    target: { value: filter },
+  });
+  await waitFor(() => {
```

Comment:
My personal preference would be that rather than using mock timers, we use the apollo helpers, eg https://github.com/strimzi/strimzi-ui/blob/a4c0f81b856c372d2e17d7e4a2194803ccd81d9f/client/Contexts/ConfigFeatureFlag/ConfigFeatureFlag.spec.tsx#L290

Reply:
Agreed.